### PR TITLE
[WIP] Feat/infra-pipelines-integration-tests

### PR DESCRIPTION
**Pull request by chris-ingrisani**
Hi, this PR adds an integration test for new SA editor role and adds an integration test for Cloud Build triggers/config #337 

- Test still needed for runner image

### CODE REVIEW

File: test/integration/projects/controls/gcloud_projects.rb [hunk 1]
```diff
@@ -191,7 +191,7 @@
             expect(data['bindings'][0]['members']).to include(
               "serviceAccount:#{cloudbuild_sa[business_unit]}"
             )
-            expect(data['bindings'][0]['role']).to eq "roles/iam.serviceAccountTokenCreator"
+            expect(data['bindings'][0]['role']).to eq ["roles/iam.serviceAccountTokenCreator", "roles/editor"]
```
Comment (bharathkkb) [inline test/integration/projects/controls/gcloud_projects.rb:194]: We should actually check for "roles/editor" on the SA project factory creates `#{base_project_sa[environment_code][business_unit]}`

File: test/integration/projects/controls/gcloud_cloudbuild.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cloudbuild_project_id = attribute('cloudbuild_project_id')
+branches_regex = '^(development|non\\-production|production)$'
+cloud_source_repos = [
+  'app_infra_repo'
+]
+
+control 'gcloud_cloudbuild' do
```
Comment (bharathkkb) [inline test/integration/projects/controls/gcloud_cloudbuild.rb:21]: We should add this control here https://github.com/terraform-google-modules/terraform-example-foundation/blob/a3a413d623d7e5a312dcfa84a67e3ff528e14abf/.kitchen.yml#L114

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @chris-ingrisani

File: test/integration/projects/controls/gcloud_projects.rb (hunk 1 above)
Comment (chris-ingrisani) [reply test/integration/projects/controls/gcloud_projects.rb:194]: Got it, thanks @bharathkkb, I'll update this.

**Pull request closed without merging**